### ISSUE TITLE
findings not reported until seemingly unrelated code is changed

### ISSUE THREAD

**Issue by firewave** [pylint-dev/pylint#10113]
### Bug description

```python
# pylint: disable=missing-module-docstring
import subprocess

def __func():
    comm = None  # disable for warning in f() to appear
    # pylint: disable-next=consider-using-with
    p = subprocess.Popen('')
    try:
        comm = p.communicate(timeout=10)
    except subprocess.TimeoutExpired:
        pass
    stdout, _ = comm  # might be None
    return stdout

# pylint: disable-next=missing-function-docstring
def f():
    s = __func()
    pos = s.find(" ")
    if not pos == -1:  # unnecessary-negation
        pass


No findings are reported for `f()` although it contains an `unnecessary-negation`. This is not limited to this warning but any findings in that function.
```

### Configuration

```ini

```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
no findings reported
```

### Expected behavior

```
************* Module test
test.py:19:7: C0117: Consider changing "not pos == -1" to "pos != -1" (unnecessary-negation)
```

I would have also expected a warning about the unpacking of `comm` potentially not having enough values.

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.5
Python 3.12.7 (main, Oct  1 2024, 11:15:50) [GCC 14.2.1 20240910]
```

### OS / Environment

Manjaro Linux, bash
Ubuntu 22.04, bash

### Additional dependencies

```python

```
